ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx

```diff
@@ -419,6 +425,46 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
       })
     );
   };
+
+  const onPreview = (item: DetailedItem) => {
+    if (item.systemType === 'component' || item.systemType === 'taxonomy') {
+      const src = `${legacyFormSrc}site=${site}&path=${item.path}&type=form&readonly=true`;
```

Comment:
We should clean up all over moving this url construction to a single place. Possibly an epic or action builder or an util that one of those would invoke.